sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -134,37 +144,168 @@ export class IER12ProcessingService {
-      mandatoryFees: offering.mandatoryFees,
-      exceptionExpenses: offering.exceptionalExpenses,
-      totalFundedWeeks: offering.studyBreaks.totalFundedWeeks,
-      disbursementSchedules: pendingAssessment.disbursementSchedules,
+    const address = student.contactInfo.address;
+    const applicationProgramYear = application.programYear;
+    const [scholasticStanding] = application.studentScholasticStandings;
+    const assessmentAwards = this.getAssessmentAwards(
+      pendingAssessment.disbursementSchedules,
+    );
+    const disbursementSchedules = pendingAssessment.disbursementSchedules;
+    const addressInfo: IERAddressInfo = {
+      addressLine1: address.addressLine1,
+      addressLine2: address.addressLine2,
+      city: address.city,
+      provinceState: address.provinceState,
+      postalCode: address.postalCode,
     };
+    const hasRestriction = this.checkActiveRestriction(
+      student.studentRestrictions,
+    );
+    const hasProvincialDefaultRestriction = !hasRestriction
+      ? false
+      : this.checkActiveRestriction(student.studentRestrictions, {
+          restrictionCode: "B2",
+        });
+    const hasPartner =
+      pendingAssessment.workflowData.studentData.relationshipStatus ===
+      "married";
+    const overawardBalance =
+      await this.disbursementOverawardService.getOverawardBalance([student.id]);
+    const studentOverawardsBalance = overawardBalance[student.id];
+    const ier12Records: IER12Record[] = [];
+    // Create IER12 records per disbursement.
+    for (const disbursement of disbursementSchedules) {
+      const [disbursementReceipt] = disbursement.disbursementReceipts;
+      const ier12Record: IER12Record = {
+        assessmentId: pendingAssessment.id,
+        disbursementId: disbursement.id,
+        applicationNumber: application.applicationNumber,
+        institutionStudentNumber: application.data.studentNumber,
+        sin: sinValidation.sin,
+        studentLastName: user.lastName,
+        studentGivenName: user.firstName,
+        studentBirthDate: new Date(student.birthDate),
+        dependantStatus:
+          pendingAssessment.workflowData.studentData.dependantStatus,
+        addressInfo,
+        programName: educationProgram.name,
+        programDescription: educationProgram.description,
+        credentialType: educationProgram.credentialType,
+        fieldOfStudyCode: educationProgram.fieldOfStudyCode,
+        currentProgramYear: offering.yearOfStudy,
+        cipCode: educationProgram.cipCode,
+        nocCode: educationProgram.nocCode,
+        sabcCode: educationProgram.sabcCode,
+        institutionProgramCode: educationProgram.institutionProgramCode,
+        programCompletionYears: educationProgram.completionYears,
+        studyStartDate: new Date(offering.studyStartDate),
+        studyEndDate: new Date(offering.studyEndDate),
+        tuitionFees: offering.actualTuitionCosts,
+        booksAndSuppliesCost: offering.programRelatedCosts,
```

Comment:
This means the programRelatedCosts has been renamed to booksAndSuppliesCost but from the model it will be assigned from programRelatedCosts right?

Reply:
Yes that is right.